### PR TITLE
docs: add backend guide with DBaaS

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: docs/guides/backend_guide.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    |         |   "serial": 0,                                          +
+    |         |   "lineage": "07a1e05b-3cba-438a-0c70-3ec5e73d4baf",    +
+    |         |   "outputs": {},                                        +
+    |         |   "resources": [                                        +
+    |         |     {
+    
+    ....
+```
+
+## Locking your backend
```
[the remark this answers is not in the document]
Comment (Monitob) [reply docs/guides/backend_guide.md:116]: I've just added a new section

File: docs/guides/backend_guide.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+page_title: "Using Backend Guide"
+description: |-
+Using your Scaleway DBaaS as to store your terraform state.
+---
+
+# Set up your database with the Scaleway provider
+
+Before initializing the backend with `terraform init`, the database must already exist:
```
Comment (remyleone) [inline docs/guides/backend_guide.md:9]: You should start with something like:

First, let's create a state locally, then we will transfer the state to a remote database.
If you already have a remote database existing you can already go to step X

File: docs/guides/backend_guide.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+Terraform provides the option to set up a [“backend”](https://www.terraform.io/language/settings/backends/configuration) of the `state` data files.
+
+This option allows you handling the state and the way certain operations are executed.
+
+Backends can store the state remotely and protect it with locks to prevent corruption;
+it makes it possible for a team to work with ease, or, for instance, to run Terraform within a pipeline.
+
+## Create your database
+
+You can create your database resource using terraform itself .
```
Comment (remyleone) [inline docs/guides/backend_guide.md:20]:
```suggestion
You can create your database resource using terraform itself.
```

File: docs/guides/backend_guide.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+provider "scaleway" {
+  # ...
+}
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# CREATE AN DATABASE INSTANCE TO USE IT AS A TERRAFORM BACKEND
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+resource "scaleway_rdb_database" "database" {
+  name        = "your-database"
```
Comment (remyleone) [inline docs/guides/backend_guide.md:43]:
```suggestion
  name        = "terraform_backend"
```

If you want to stay consistent with the URL down below

File: docs/guides/backend_guide.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+Check your database `schema`. e.g:
+
+```sql
+rdb=> SELECT * FROM pg_catalog.pg_tables WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema';
+       schemaname       | tablename | tableowner | tablespace | hasindexes | hasrules | hastriggers | rowsecurity
+------------------------+-----------+------------+------------+------------+----------+-------------+-------------
+ terraform_remote_state | states    | my_initial_user |            | t          | f        | f           | f
+```
+
+After running terraform `apply`, to terraform.tfstate on the database will look something like this:
```
Comment (remyleone) [inline docs/guides/backend_guide.md:112]:
```suggestion
After running terraform `apply`, to terraform. The state on the database will look something like this:
```

File: docs/guides/backend_guide.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+```shell
```
Comment (remyleone) [inline docs/guides/backend_guide.md:114]: this is not shell language. Just use simple text

File: docs/guides/backend_guide.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+Answer the prompt `yes`, and your state will migrate.
+
+```shell
+$ terraform init -backend-config="conn_str=${PG_CONN_STR}" -migrate-state
+```
+
+## What about locking?
+
+Most of the remote backends natively support locking. To run terraform apply, Terraform will automatically acquire a lock;
```
Comment (remyleone) [inline docs/guides/backend_guide.md:166]: Add a link to the documentation of the different available backends

File: docs/guides/backend_guide.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+You can run apply with the `-lock-timeout=<TIME>` parameter to tell Terraform to wait up to TIME for a lock to be released (e.g., `-lock-timeout=10m` will wait for 10 minutes).
+
+The Lock method, prevents opening the state file while already in use.
+
+## Share configuration
+
+You can also share the configuration using the different [data sources](https://www.terraform.io/language/state/remote-state-data).
+this is useful when you are working on the same infrastructure we the same team.
+
+For further information about thea advantages of t
```
Comment (remyleone) [inline docs/guides/backend_guide.md:177]: seems like the text is not finished

File: docs/guides/backend_guide.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+```hcl
+data "scaleway_rdb_instance" "mybackend" {
+    name = "your-database-name"
+}
+```
+
+## Limitation
+The Scaleway Object Storage based backends is not supported by scaleway.
```
Comment (remyleone) [inline docs/guides/backend_guide.md:186]: This sentence needs to be reworked. Only locking is not supported, but storing of the state is

File: docs/guides/backend_guide.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+Using your Scaleway DBaaS as to store your terraform state.
+---
+
+# Terraform Backend
+
+This page describes how to configure a backend by adding the backend block to your configuration with the Terraform Scaleway Provider.
+
+Terraform provides the option to set up a [“backend”](https://www.terraform.io/language/settings/backends/configuration) of the `state` data files.
+
+This option allows you handling the state and the way certain operations are executed.
```
Comment (Vinzius) [inline docs/guides/backend_guide.md:13]:
```suggestion
This option allows you to handle the state and the way certain operations are executed.
```

File: docs/guides/backend_guide.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+
+and deploy it:
+
+```shell
+terraform plan -out "planfile" ; terraform apply -input=false -auto-approve "planfile"
+```
+
+## Configuring the PostgreSQL Connection String
+
+We choose to set our environment variable for the connection string for this guide. Please check the [secret section](#secrets) for mor details.
```
Comment (Vinzius) [inline docs/guides/backend_guide.md:67]:
```suggestion
We choose to set our environment variable for the connection string for this guide. Please check the [secret section](#secrets) for more details.
```

File: docs/guides/backend_guide.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+```shell
+ export PG_CONN_STR=postgres://<user>:<pass>@localhost:<port>/terraform_backend?sslmode=disable
+```
+
+## Secrets
+
+Hashicorp propose several methods to keep your secrets. Please check the Terraform  [partial configuration](https://www.terraform.io/language/settings/backends/configuration#partial-configuration) for this topic.
```
Comment (Vinzius) [inline docs/guides/backend_guide.md:75]:
```suggestion
Hashicorp proposes several methods to keep your secrets. Please check the Terraform  [partial configuration](https://www.terraform.io/language/settings/backends/configuration#partial-configuration) for this topic.
```

File: docs/guides/backend_guide.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+$ terraform init -backend-config="conn_str=${PG_CONN_STR}" -migrate-state
+```
+
+## What about locking?
+
+Most of the remote [backends](https://www.terraform.io/language/settings/backends/configuration#available-backends) natively support locking. To run terraform apply, Terraform will automatically acquire a lock;
+if someone else is already running apply, they will already have the lock, and you will have to wait.
+You can run apply with the `-lock-timeout=<TIME>` parameter to tell Terraform to wait up to TIME for a lock to be released (e.g., `-lock-timeout=10m` will wait for 10 minutes).
+
+The Lock method, prevents opening the state file while already in use.
```
Comment (Vinzius) [inline docs/guides/backend_guide.md:170]:
```suggestion
The Lock method prevents opening the state file while already in use.
```

File: docs/guides/backend_guide.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+
+## Share configuration
+
+You can also share the configuration using the different [data sources](https://www.terraform.io/language/state/remote-state-data).
+this is useful when you are working on the same infrastructure we the same team.
```
Comment (Vinzius) [inline docs/guides/backend_guide.md:175]:
```suggestion
You can also share the configuration using the different [data sources](https://www.terraform.io/language/state/remote-state-data).
This is useful when working on the same infrastructure or the same team.
```

File: docs/guides/backend_guide.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+```shell
+terraform plan -out "planfile" ; terraform apply -input=false -auto-approve "planfile"
+```
+
+## Configuring the PostgreSQL Connection String
+
+We choose to set our environment variable for the connection string for this guide. Please check the [secret section](#secrets) for more details.
+
+```shell
+ export PG_CONN_STR=postgres://<user>:<pass>@localhost:<port>/terraform_backend?sslmode=disable
```
Comment (remyleone) [inline docs/guides/backend_guide.md:70]:
```suggestion
export PG_CONN_STR=postgres://<user>:<pass>@localhost:<port>/terraform_backend?sslmode=disable
```

File: docs/guides/backend_guide.md [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+## Secrets
+
+Hashicorp proposes several methods to keep your secrets. Please check the Terraform  [partial configuration](https://www.terraform.io/language/settings/backends/configuration#partial-configuration) for this topic.
```
Comment (remyleone) [inline docs/guides/backend_guide.md:75]:
```suggestion
Hashicorp offers several methods to keep your secrets. Please check the Terraform [partial configuration](https://www.terraform.io/language/settings/backends/configuration#partial-configuration) for this topic.
```

File: docs/guides/backend_guide.md [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+You can also share the configuration using the different [data sources](https://www.terraform.io/language/state/remote-state-data).
+This is useful when working on the same infrastructure or the same team.
+
+```hcl
+data "scaleway_rdb_instance" "mybackend" {
+    name = "your-database-name"
+}
+```
+
+## Limitation
```
Comment (remyleone) [inline docs/guides/backend_guide.md:183]: remove this section completly

**Pull request merged**